### ISSUE TITLE
JPQL: 'count' not working for mongodb

### ISSUE THREAD

**Issue by dreamingo88** [Impetus/kundera#886]
1. persistence.xml:

```
<persistence-unit name="MONGO_NTV">
    <provider>com.impetus.kundera.KunderaPersistence</provider>
    <class>models.Category</class>
    <exclude-unlisted-classes>true</exclude-unlisted-classes>
    <properties>
        <property name="kundera.nodes" value="localhost" />
        <property name="kundera.port" value="27017" />
        <property name="kundera.keyspace" value="xplayks" />
        <property name="kundera.dialect" value="mongodb" />
        <property name="kundera.client.lookup.class" value="com.impetus.client.mongodb.MongoDBClientFactory" />
        <property name="kundera.ddl.auto.prepare" value="update" />
        <property name="kundera.show.query" value="true" />
    </properties>
</persistence-unit>
```
1. Category.java:

```
@Id
@Column(name="id")
public String id;

@Column(name="name")
public String name;

@Column(name="code")
public String code;

@Column(name="description")
public String description;
```
1. createCategory() using MONGO_PU:

```
for(int i = 9000; i < 9030; i++) {
    Category ctg = new Category();
    ctg.name = "just a simple demo " + i;
    ctg.code = "democode";
    ctg.description = "just a demo, using mongodb";

    em.persist(ctg);
}
```
1. findCategory() using MONGO_NTV:

```
Query query = em.createQuery("select count(ctg) from Category ctg where ctg.code = 'democode'");
String count = query.getResultList().get(0).toString();
System.out.println("================="+count); //print 'just a simple demo ...', it's very very strange, the right value should be 30, 
```

Question: what's the problem? 'count' not supported by mongodb?

Tips: I using kundera 3.6 with mongodb 3.2.

**Comment by devender-yadav** [comment]
@dreamingo88 

Count is not supported via JPQL. 

You can run native mongo query via `executeScript()` client method for this. 

Sample code:

```
    Map<String, Client<Query>> clients = (Map<String, Client<Query>>) em.getDelegate();
    Client client = clients.get("MONGO_NTV");
    script = "db.category.count({code:'democode'})";
    result = (client).executeScript(script);
    long totalDocuments = ((Double) result).longValue();
    Assert.assertEquals(30, totalDocuments);
```

Check [testcase](https://github.com/impetus-opensource/Kundera/blob/6ee77aa5ee891be90d181ec41c52c1af8cacdb5f/src/kundera-mongo/src/test/java/com/impetus/client/crud/PersonMongoTest.java#L319) for more details.

-Dev

**Comment by dreamingo88** [comment]
@devender-yadav 

Is there any plan to support it via JPQL?
It's very important and useful.

**Comment by devender-yadav** [comment]
@dreamingo88 

Did the solution provided by me work for you?

We can pick this feature in future releases but there is no plan for this in the next release.

-Dev

**Comment by dreamingo88** [comment]
@devender-yadav 

Yes, It works right via native way.

But I still hope this feature can support via JPQL.

The most important reason to use JPA is: Do not need to pay attention to the difference of each query language, only need to master JPQL.

Thanks for your hard work!

**Comment by devender-yadav** [comment]
@dreamingo88 

You're welcome. Happy to help.

I agree with you on supporting basic feature via JPQL. We will take this in future releases.

-Dev

**Comment by devender-yadav** [comment]
Hi @dreamingo88,

We have added support for COUNT query via JPQL . It will be available from the next release (_Kundera 3.9_).
Check [test-case](https://github.com/impetus-opensource/Kundera/blob/c4e3586e0c49a3110baaae8f23cbe7167b65263b/src/kundera-mongo/src/test/java/com/impetus/client/crud/PersonMongoTest.java#L930) for more details.


-Dev
